### PR TITLE
Convert contract transactions from old persisted format

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/164113577

### CODE REVIEW

File: apps/aecontract/src/aect_create_tx.erl [hunk 1]
```diff
@@ -72,6 +77,15 @@
 
 -export_type([tx/0]).
 
+%%%===================================================================
+%%% Conversion of old db format
+
+-spec from_db_format(tx()) -> tx().
+from_db_format(#contract_create_tx{ct_version = #{}} = Tx) ->
+    Tx;
+from_db_format(#contract_create_tx{ct_version = VMVersion} = Tx) ->
```
Comment (lucafavatella) [inline apps/aecontract/src/aect_create_tx.erl]: Guard on type of VMVersion would have made this more readable though minor and debatable. Ok.

File: apps/aecontract/src/aect_call_tx.erl [hunk 2]
```diff
@@ -104,6 +123,46 @@ new(#{caller_id   := CallerId,
...
+           contract_id      = ContractId,
+           abi_version      = ABI,
+           fee              = Fee,
+           ttl              = TTL,
+           amount           = Amount,
+           gas              = Gas,
+           gas_price        = GasPrice,
+           call_data        = CallData,
+           call_stack       = CallStack
+          } ->
```
Comment (lucafavatella) [inline apps/aecontract/src/aect_call_tx.erl]: So the absence of field `call_origin` in record `db_contract_call_tx` is used as signal for detecting old version. Ok.
I see not int/map different representation for any version field so no transformation required for any field. Ok.

File: apps/aecontract/src/aect_call_tx.erl [hunk 3]
```diff
@@ -104,6 +123,46 @@ new(#{caller_id   := CallerId,
...
+               fee              = Fee,
+               ttl              = TTL,
+               amount           = Amount,
+               gas              = Gas,
+               gas_price        = GasPrice,
+               call_data        = CallData,
+               call_stack       = CallStack,
+               call_origin      = CallOrigin
+              };
+        _ ->
```
Comment (ThomasArts) [inline apps/aecontract/src/aect_call_tx.erl]: is there a risk that one would call this function once more after the first conversion and then fail with an error?

Comment (gorillainduction) [reply apps/aecontract/src/aect_call_tx.erl]: It will then be caught by the previous function clause.

**Pull request merged**